### PR TITLE
Styling of mobile header consistant with desktop version

### PR DESCRIPTION
**Pull request by ghost**
Not sure with the drawer background color. Tell me what you think.

screen shot 2015-08-02 at 21 37 10

screen shot 2015-08-02 at 21 37 14

### CODE REVIEW

**Comment by svbergerem** [comment]
I think I'd increase the font-size for those icons.

File: app/views/layouts/application.mobile.haml [hunk 1]
```diff
@@ -52,25 +52,7 @@
...
-                  = current_user.unread_message_count
-            -# Publisher
-            = link_to(image_tag("mobile/compose_mobile.png"), new_status_message_path, class: "badge", id: "compose_badge")
-            -# Menu
-            = link_to(image_tag("mobile/menu.png"), "#", id: "menu_badge", class: "badge")
-          = link_to(image_tag("mobile/asterisk_white_mobile.png"), stream_path, id: "header_title")
+        = render "layouts/header"
 
       - if user_signed_in?
         #drawer
```
Comment (svbergerem) [inline app/views/layouts/application.mobile.haml]: I guess you wanted to use `layouts/drawer` here which you already wrote. You could also remove the `if user_signed_in?` line and put the new line to render the drawer directly under the one to render the header.

**Comment by AugierLe42e** [comment]
Yeah, me too. It did't render the same way on the big screen :p

File: app/views/layouts/application.mobile.haml (hunk 1 above)
Comment (AugierLe42e) [inline app/views/layouts/application.mobile.haml]: Oops :)

File: app/assets/stylesheets/mobile/header.scss [hunk 2]
```diff
@@ -83,16 +84,13 @@ header {
 
 
         input {
-          box-shadow: 0 1px 1px #444;
-          border-radius: 15px;
           margin-top: 5px;
```
Comment (svbergerem) [inline app/assets/stylesheets/mobile/header.scss]: Try 7px here. That moves the search input to the middle of the header.

Comment (AugierLe42e) [inline app/assets/stylesheets/mobile/header.scss]: Right, thx, I've forgot this.

**Comment by AugierLe42e** [comment]
This should build and be ready to merge, now. How it looks:

screen shot 2015-08-03 at 17 38 48

screen shot 2015-08-03 at 17 38 50

**Comment by AugierLe42e** [comment]
I have changed the icon used for writing a message from `.entypo-pencil` to Bootstrap's `..glyphicon-edit` as I jut remarked that `.entypo-pencil` is already used as icon to edit aspect name. Here is how it looks:

screen shot 2015-08-03 at 19 01 30

**Comment by Flaburgan** [comment]
I'll review and test this soon.

File: app/assets/stylesheets/mobile/header.scss [hunk 3]
```diff
@@ -118,16 +136,16 @@ header {
       line-height: 25px;
       font-weight: bold;
       color: $light-grey;
-      border-bottom: solid rgb(53, 53, 53) 2px;
+      border-bottom: solid $light-grey 1px;
```
Comment (Flaburgan) [inline app/assets/stylesheets/mobile/header.scss]: light-grey is #ddd, looks like you did a mistake here. And indeed, the borders are #ddd so almost white for me, not #353535 as actually (so your screenshot is not up-to-date). This color doesn't exist in our variables currently. $border-dark-grey is #999 which is still to light imo. Maybe it's worth creating a new variable?

**Comment by Flaburgan** [comment]
The border color of the search field is also wrong, `#999` in this PR when `#222` previously and `#777` in the desktop view. I suggest you to use the same color here than for [the li border](https://github.com/diaspora/diaspora/pull/6285/files#diff-f1d87b0373e1e27ac198707ab6fae6e3R139) so possibly the new variable created. Also, I find that the input field is almost invisible, could you please restore the previous color background? This could also be done in the desktop version imo.

**Comment by Flaburgan** [comment]
The header icons can still be bigger. They are smaller and closer than before, and there should not be a border for the drawer icon for consistency.

screen

Also, I'm all with you when it's about unification and consistency between desktop and mobile, but I honestly find that those icons are not as beautiful as the current ones. I don't know what we want to do about that.

File: app/assets/stylesheets/mobile/header.scss [hunk 4]
```diff
@@ -2,56 +2,75 @@
 
 $drawer-width: 400px;
 $drawer-width-offset: $drawer-width + 50px;
+$mobile-navbar-height: 46px;
 
-header {
-  position: fixed;
```
Comment (Flaburgan) [inline app/assets/stylesheets/mobile/header.scss]: Please keep the header fixed at the top of the screen. It now disappears when we scroll.

File: app/assets/stylesheets/mobile/header.scss (hunk 3 above)
Comment (AugierLe42e) [inline app/assets/stylesheets/mobile/header.scss]: Nope, it's no mistake. The color here is the same color than search box border.

**Comment by AugierLe42e** [comment]
> The border color of the search field is also wrong, `#999` in this PR when `#222` previously and `#777` in the desktop view. I suggest you to use the same color here than for the li border so possibly the new variable created. Also, I find that the input field is almost invisible, could you please restore the previous color background? This could also be done in the desktop version imo.

> Also, I'm all with you when it's about unification and consistency between desktop and mobile, but I honestly find that those icons are not as beautiful as the current ones.

As you might have read in the title, this PR is about making header compilant with desktop version. If you want design changes, it's ok but it is out of the scope of this PR. Please stick to technical points ;)

**Comment by Flaburgan** [comment]
>As you might have read in the title, this PR is about making header compilant with desktop version. If you want design changes, it's ok but it is out of the scope of this PR. Please stick to technical points ;)

I don't agree. To be consistent visually is a nice goal, but it **is** a design change. The render is not the same. So I review this as a design change. And comparing what we have now and what you propose, I definitely prefer what we have now:
capture d ecran 2015-08-06 a 10 46 01

capture d ecran 2015-08-06 a 10 45 20

File: app/assets/stylesheets/mobile/header.scss (hunk 4 above)
Comment (AugierLe42e) [inline app/assets/stylesheets/mobile/header.scss]: Done.

**Comment by AugierLe42e** [comment]
> I don't agree. To be consistent visually is a nice goal, but it is a design change. The render is not the same. So I review this as a design change. And comparing what we have now and what you propose, I definitely prefer what we have now:

I don't care. I'm tired facing "I can has" request when I propose PRs. @svbergerem redesigned the header and you actually liked it. I'm just affecting these changes on the mobile version. If you don't like them and you don't mind make the project look amateurish, talk about to @svbergerem or open a Loomio discussion.

**Comment by Flaburgan** [comment]
This is not about the header but about the drawer. You want to make the desktop and the mobile views consistent? Then where in any dropdown do you see a white separator like this? And as said, the search border color is not even the one used in the desktop version.

File: app/assets/stylesheets/mobile/header.scss [hunk 5]
```diff
@@ -2,56 +2,75 @@
...
-      display: inline;
-      margin: 0 4px;
-      padding: 10px 6px;
-      font-weight: bold;
-      font-size: smaller;
+    li > a, li > button {
       background-color: transparent;
+
+      &.badge-link {
+        font-size: 24px;
```
Comment (svbergerem) [inline app/assets/stylesheets/mobile/header.scss]: I thought you already set it to 26px?!

**Comment by AugierLe42e** [comment]
> Then where in any dropdown do you see a white separator like this?

Around the input field. Where in any dropdown do you see a gray separator like this?



This discussion is pointless. The border is the only real design changed I made, but you asked me to change:

 1. the icons,
 2. the input field,
 3. the borders.

The reason there is border on mobil version and not on desktop is that you can *hover* the elements on desktop to highlight where you click. Not on a smartphone.

We can discuss about the size and the color of the icons, the backgrounds, etc. *only* if this is changed both on desktop and on mobile. Otherwirse, I won't change the icons and I wont change the input field.

**Comment by svbergerem** [comment]
@AugierLe42e already increased the font-size of the icons but somehow reverted the change. I prefer to have the same icons on desktop and mobile but I am not 100% sure about the new post icon. I agree with @Flaburgan that those white borders in the drawer look bad and there is no similar element on the desktop with those borders. I'd like to keep the amount of used colors as small as possible (consistency-foo ;)). Maybe there is a color in `bootstrap-variables.scss` which is ok?

About the search input: on the desktop it is always visible and having a white input in the dark header would be distracting. (IMO) We want to focus on the content (posts etc.) and I think a light background would be bad in that case. When the user opens the drawer on the mobile view then the posts aren't that important anymore so we could use a light background in the mobile version to improve the usability.

**Comment by AugierLe42e** [comment]
> @AugierLe42e already increased the font-size of the icons but somehow reverted the change.

Not really. It was 20px at the beginning. Then I increased to 26px. Now it's 24px and 20px for `entypo-bell` because it was too big. I've had a hard time choosing the sizes :/

>  I am not 100% sure about the new post icon.

Nope, me neither but I can't find something that fits well otherwise.

> and there is no similar element on the desktop with those borders.

The search bar ;)

> I think a light background would be bad in that case.

I'll find weird not to have the same element. Everything else is the same than desktop but not this? This woul be bizarre as it really appears as a design choice more than an accessibility one. :/

**Comment by AugierLe42e** [comment]
About the icons, a radical change could be considering moving to [font-awsome](https://fortawesome.github.io/Font-Awesome/icons/) which seems to have much more icons.

**Comment by AugierLe42e** [comment]
@svbergerem: I reverted the changes of the icons font size. Here is how it looks now:

screen shot 2015-08-06 at 20 42 08

**Comment by AugierLe42e** [comment]
FYI, if I add `.navbar-inverse` to drawer too, here is how it looks:

screen shot 2015-08-06 at 20 49 23

As there is a good contrast, removing the `font-weight: borld` is better:

screen shot 2015-08-06 at 20 49 36

I really like this last one. Very sexy.

Or we can use the default `border-color` defined for `navbar-inverse`. I like it too:

screen shot 2015-08-06 at 20 53 38

**Comment by AugierLe42e** [comment]
So? What's next?

**Comment by AugierLe42e** [comment]
About the icons, we could use [fontforge](http://fontcustom.com/) to design custom icons.

**Comment by AugierLe42e** [comment]
I changed the drawer to be black like on the last picture I posted. It seems it's a go for @Flaburgan and I like it too. I propose to merge it as is and open a new PR to discuss about the icons. I liked the compose and mail images so what I propose is to extact these images as SVG fonts and then convert them to TTF files.

**Comment by AugierLe42e** [comment]
I've converted the previous images to proper font icons. Here is how it looks now:

screen shot 2015-08-14 at 17 57 55

screen shot 2015-08-14 at 17 57 57

**Comment by jaywink** [comment]
:+1: from me from the last versions, looks good.

**Comment by AugierLe42e** [comment]
Nice! Thanks @jaywink. As this seems to be ok with everyone, I suggest someone do a final review of it?

File: app/views/layouts/_drawer.mobile.haml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+#drawer
+  %header
+    #global-search
+      =form_tag("/search", method: "get", class: "search_form", "accept-charset" => "UTF-8") do
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: Style/HashSyntax: Don't mix styles in the same hash.

File: app/views/layouts/_drawer.mobile.haml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        %div
+          =hidden_field_tag "utf8", "✓"
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+          =search_field_tag "q", nil, id: "q", placeholder: t("search"), results: "5", autocomplete: "off", class: "ac_input form-control"
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: Line is too long. [138/120]

File: app/views/layouts/_drawer.mobile.haml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  %nav.navbar-inverse
+    %ul
+      %li=link_to t("streams.activity.title"), activity_stream_path
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+      %li=link_to t("streams.mentions.title"), mentioned_stream_path
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+      %li#all_aspects
+        =link_to t("streams.aspects.title"), "#"
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+      %li.no-border.hide
+        %ul
+          -current_user.aspects.each do |aspect|
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The - symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+            %li=link_to aspect.name, aspects_stream_path(a_ids: [aspect.id])
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+      %li#followed_tags
+        =link_to t("streams.followed_tag.title"), "#"
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+      %li.no-border.hide
+        %ul
+          -current_user.followed_tags.each do |tag|
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The - symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+            %li=tag_link(tag)
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+          -if current_user.followed_tags.length > 0
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The - symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+            %li.manage-followed-tags
+              =link_to t("tag_followings.manage.title"), manage_tag_followings_path
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+      %li
+        =link_to user_profile_path(current_user.username) do
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+          =t("layouts.header.profile")
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+          =person_image_tag(current_user, size: :thumb_small)
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+      %li=link_to t("_contacts"), contacts_path
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+      %li=link_to t("layouts.header.settings"), users_edit_path
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 24]
```diff
@@ -0,0 +1,34 @@
...
+      %li=link_to t("layouts.application.toggle"), toggle_mobile_path
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_drawer.mobile.haml [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+      %li=link_to t("layouts.header.logout"), destroy_user_session_path, method: :delete
```
Comment (diaspora-code-review) [inline app/views/layouts/_drawer.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_header.mobile.haml [hunk 26]
```diff
@@ -0,0 +1,35 @@
+.nav.navbar-inverse.navbar-fixed-top#main-nav
+  .container-fluid
+    .navbar
+      =link_to(image_tag("mobile/asterisk_white_mobile.png", class: "img-responsive"),
+          stream_path, id: "header-title", class: "navbar-brand")
+
+      %ul#nav-badges.nav.navbar-nav
```
Comment (diaspora-code-review) [inline app/views/layouts/_header.mobile.haml]: Classes should be listed before IDs (.nav should precede #nav-badges)

File: app/views/layouts/_header.mobile.haml [hunk 27]
```diff
@@ -0,0 +1,35 @@
...
+        -# Notifications
+        %li
+          =link_to notifications_path, class: "badge-link", id: "notification-badge" do
+            %i.entypo-bell
+            -if current_user.unread_notifications.size > 0
```
Comment (diaspora-code-review) [inline app/views/layouts/_header.mobile.haml]: The - symbol should have one space separating it from code

File: app/views/layouts/_header.mobile.haml [hunk 28]
```diff
@@ -0,0 +1,35 @@
...
+              %span.badge{id: "notification"}
+                =current_user.unread_notifications.size
```
Comment (diaspora-code-review) [inline app/views/layouts/_header.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_header.mobile.haml [hunk 29]
```diff
@@ -0,0 +1,35 @@
...
+
+        -# Conversations
+        %li
+          =link_to conversations_path, class: "badge-link", id: "conversations-badge" do
```
Comment (diaspora-code-review) [inline app/views/layouts/_header.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_header.mobile.haml [hunk 30]
```diff
@@ -0,0 +1,35 @@
...
+            %i.diaspora-custom-mail
+            -if current_user.unread_message_count > 0
```
Comment (diaspora-code-review) [inline app/views/layouts/_header.mobile.haml]: The - symbol should have one space separating it from code

File: app/views/layouts/_header.mobile.haml [hunk 31]
```diff
@@ -0,0 +1,35 @@
...
+              %span.badge{id: "conversation"}
+                =current_user.unread_message_count
```
Comment (diaspora-code-review) [inline app/views/layouts/_header.mobile.haml]: The = symbol should have one space separating it from code

File: app/views/layouts/_header.mobile.haml [hunk 32]
```diff
@@ -0,0 +1,35 @@
...
+
+        -# Publisher
+        %li
+          =link_to new_status_message_path, class: "badge-link", id: "compose-badge" do
+            %i.diaspora-custom-compose
+
+        -# Menu
+        %li
+          %button#menu-badge.navbar-toggle{type: "button"}
```
Comment (diaspora-code-review) [inline app/views/layouts/_header.mobile.haml]: Classes should be listed before IDs (.navbar-toggle should precede #menu-badge)

File: app/assets/stylesheets/mobile/mobile.scss [hunk 33]
```diff
@@ -34,12 +36,16 @@ body {
 
 h3 {  margin-top: 0; }
 .clear { clear: both; }
-#main { padding: 55px 10px 0 10px; }
+#main { padding: 56px 10px 0 10px;; }
```
Comment (svbergerem) [inline app/assets/stylesheets/mobile/mobile.scss]: Superfluous semicolon.

Comment (AugierLe42e) [inline app/assets/stylesheets/mobile/mobile.scss]: Nice catch !

**Comment by svbergerem** [comment]
Thank you! :)

**Pull request merged**